packages/server/validators/auth.js

```diff
@@ -0,0 +1,52 @@
+const { body } = require("express-validator");
+const AuthController = require("../controllers/auth");
+const UserController = require("../controllers/users");
+const { checkInvalidPasswordResetToken } = require("../utils/redis");
+
+const forgotPasswordValidator = [
+  body("email", "Invalid email.")
+    .trim()
+    .exists({ checkFalsy: true })
+    .withMessage("Email can't be null or empty.")
+    .isEmail()
+    .withMessage("Invalid email entered!")
+
+    // Check If User Exists in DB & If User is a Google OAuth Acc
+    .custom(async (email) => {
+      const userExists = await AuthController.isUserEmail(email);
+      if (!userExists) {
+        throw new Error("User with given email does not exist.");
+      }
+
+      const userId = await UserController.getUserIdByEmail(email);
+      const isGoogleAccount = await AuthController.isGoogleAccount(userId);
+      if (isGoogleAccount) {
+        throw new Error("Please login using Google.");
+      }
+    }),
+];
+
+const passwordResetValidator = [
+  body("token", "Invalid token.")
+    .trim()
+    .exists({ checkFalsy: true })
+    .withMessage("Token can't be null or empty.")
+
+    // Check if the password reset token has been used before.
+    .custom(async (token) => {
+      const isUsedToken = await checkInvalidPasswordResetToken(token);
+      if (isUsedToken) {
+        throw new Error("This password reset token is expired/invalid!");
+      }
+    }),
+
+  body("password", "Invalid password.")
+    .trim()
+    .exists({ checkFalsy: true })
+    .withMessage("Password can't be null or empty."),
```

Comment:
We should enforce some rules for passwords. These are just my initial thoughts and we can polish this out:
- 12 to 20 characters in length
- At least 1 uppercase character
- At least 1 number
- At least 1 special character from these `!@#$%^&*`

You should use a regular expression in the express-validator to enforce this.